src/Widgets/DeviceItem.vala

```diff
@@ -123,21 +123,25 @@ namespace Network.Widgets {
                 } else {
                     subtitle = Utils.state_to_string (state);
                 }
-            } else if (custom_mode != Utils.CustomMode.INVALID) {
+            } else {
                 switch (custom_mode) {
-                    case Utils.CustomMode.PROXY_NONE:
+                    case Utils.ProxyMode.PROXY_NONE:
                         subtitle = _("Disabled");
                         status_image.icon_name = "user-offline";
                         break;
-                    case Utils.CustomMode.PROXY_MANUAL:
+                    case Utils.ProxyMode.PROXY_MANUAL:
                         subtitle = _("Enabled (manual mode)");
                         status_image.icon_name = "user-available";
                         break;
-                    case Utils.CustomMode.PROXY_AUTO:
+                    case Utils.ProxyMode.PROXY_AUTO:
                         subtitle = _("Enabled (auto mode)");
                         status_image.icon_name = "user-available";
                         break;
-               }
+                    case Utils.ProxyMode.INVALID:
+                        subtitle = _("Disabled (error)");
+                        status_image.icon_name = "user-busy";
```

Comment:
I use the "user-busy" icon to display the error, as for the failed DeviceState on line 117 above in order to be coherent.